oneoffs/prepare_bigquery.py

```diff
@@ -0,0 +1,185 @@
+        root_node, game_data['worker_id'], game_data['completed_time'],
+        game_data['board_size'])
+    return game_data, move_data
+
+def extract_game_data(gcs_path, root_node):
+    props = root_node.properties
+    komi = float(sgf_wrapper.sgf_prop(props.get('KM')))
+    result = sgf_wrapper.sgf_prop(props.get('RE', ''))
+    board_size = int(sgf_wrapper.sgf_prop(props.get('SZ')))
+    value = utils.parse_game_result(result)
+    was_resign = '+R' in result
+    
+    filename = os.path.basename(gcs_path)
+    filename_no_ext, _ = os.path.splitext(filename)
+    completion_time = int(filename_no_ext.split('-')[0])
+    worker_id = filename_no_ext.split('-')[-1]
+    model_num = shipname.detect_model_num(props.get('PW')[0])
+    sgf_url = gcs_path
+    first_comment_node_lines = root_node.next.properties['C'][0].split('\n')
+    # in-place edit to comment node so that first move's comment looks
+    # the same as all the other moves.
+    root_node.next.properties['C'][0] = '\n'.join(first_comment_node_lines[1:])
+    resign_threshold = float(first_comment_node_lines[0].split()[-1])
+
+    return {
+        'worker_id': worker_id,
+        'completed_time': completion_time * 1000, # BigQuery's TIMESTAMP() takes in unix millis.
+        'board_size': board_size,
+        'model_num': model_num,
+        'result_str': result,
+        'value': value,
+        'was_resign': was_resign,
+        'sgf_url': sgf_url,
+        'resign_threshold': resign_threshold,
+    }
+
+def extract_move_data(root_node, worker_id, completed_time, board_size):
+    current_node = root_node.next
+    move_data = []
+    move_num = 1
+    while current_node is not None:
+        props = current_node.properties
+        if 'B' in props:
+            to_play = 1
+            move_played = props['B'][0]
+        elif 'W' in props:
+            to_play = -1
+            move_played = props['W'][0]
+        else:
+            import pdb; pdb.set_trace()
+        move_played = coords.to_flat(coords.from_sgf(move_played))
+        post_Q, debug_rows = parse_comment_node(props['C'][0])
+        policy_prior = [0] * (board_size * board_size + 1)
+        policy_prior_orig = policy_prior[:]
+        mcts_visit_counts = policy_prior[:]
+        mcts_visit_counts_norm = policy_prior[:]
+        for debug_row in debug_rows:
+            move = debug_row.move
+            policy_prior[move] = debug_row.prior
+            policy_prior_orig[move] = debug_row.orig_prior
+            mcts_visit_counts[move] = debug_row.N
+            mcts_visit_counts_norm[move] = debug_row.soft_N
+
+        move_data.append({
+            'worker_id': worker_id,
+            'completed_time': completed_time,
+            'move_num': move_num,
+            'turn_to_play': to_play,
+            'move': move_played,
+            'move_kgs': coords.to_kgs(coords.from_flat(move_played)),
+            'prior_Q': None,
+            'post_Q': post_Q, 
+            'policy_prior': policy_prior,
```

Comment:
This seems to result in each Move row being 11,000 Bytes (according to BigQuery).
Consider either not including this data or including the index of the move so that you only have to store 10 policy_priors (instead of 361)

Reply:
I want to keep the schema general because I intend for the selfplay workers to emit full debug info, not just info on the top 10.